c10/core/impl/DeviceGuardImplInterface.h

```diff
@@ -291,6 +301,10 @@ struct NoOpDeviceGuardImpl : public DeviceGuardImplInterface {
     return 1;
   }
 
+  DeviceCapability getDeviceCapability(Device /*unused*/) const override {
+    return DeviceCapability();
```

Comment:
`NoOpDeviceGuardImpl` is mainly used for `CPU` and `Meta`, so it is necessary to check the dtype range supported by these two devices separately.